payjoin/src/core/receive/mod.rs

```diff
@@ -241,14 +241,14 @@ pub(crate) fn parse_payload(
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub(crate) struct PsbtContext {
+pub struct PsbtContext {
```

Comment:
The `receive::v2::SessionEvent::ProvisionalProposal(PsbtContext)` requires this visibility. It's possible to wrap this struct if desired, but PsbtContext is the actual event diff that we're using. Perhaps it could be pruned further, just to the bitcoin::Psbt that's the payjoin_psbt since we already have the original psbt at the UncheckedProposal stage.

IMO UncheckedProposal((Original, Option<crate::HpkePublicKey>)) deserves its own type as well instead of a tuple but that can come after. The `UncheckedProposal` tuple should probably at least be deconstructed out of the nested tuple to `UncheckedProposal(Original, Option<crate::HpkePublicKey>)`